docs/User/Tutorials/Introduction.md

```diff
@@ -372,90 +404,86 @@ streams concurrently.
+  line = Fold.takeEndBy (== '\n') Array.write
+
+recv :: Socket -> Stream IO (Array Char)
+recv sk = Stream.finallyIO (close sk) (readLines sk)
 
 -- | Starts a server at port 8091 listening for lines with space separated
 -- words. Multiple clients can connect to the server and send streams of lines.
 -- The server handles all the connections concurrently, merges the incoming
 -- streams at line boundaries and writes the merged stream to a file.
 server :: Handle -> IO ()
 server file =
-      Stream.unfold TCP.acceptOnPort 8090        -- SerialT IO Socket
-    & Stream.concatMapWith Stream.parallel recv  -- SerialT IO (Array Char)
-    & Stream.unfoldMany Array.read               -- SerialT IO Char
-    & Unicode.encodeLatin1                       -- SerialT IO Word8
-    & Stream.fold (Handle.write file)            -- IO ()
+      Stream.unfold TCP.acceptorOnPort 8090        -- Stream IO Socket
+    & Stream.parConcatMap (Stream.eager True) recv -- Stream IO (Array Char)
+    & Stream.unfoldMany Array.reader               -- Stream IO Char
+    & Unicode.encodeLatin1                         -- Stream IO Word8
+    & Stream.fold (Handle.write file)              -- IO ()
 
 main :: IO ()
 main = withFile "output.txt" AppendMode server
 ```
 
 ### Listing Directories Recursively/Concurrently
 
-Our next example lists a directory tree recursively, reading
-multiple directories concurrently.
-
-This example uses the tree traversing combinator `iterateMapLeftsWith`.
-This combinator maps a stream generator on the `Left` values in its
-input stream (directory names in this case), feeding the resulting `Left`
-values back to the input, while it lets the `Right` values (file names
-in this case) pass through to the output. The `Stream.ahead` stream
-joining combinator then makes it iterate on the directories concurrently.
-
-Please see the file [ListDir.hs][] for the complete working code,
-including the imports that we have omitted below.
-
-```haskell
-import Streamly.Internal.Data.Stream.IsStream (iterateMapLeftsWith)
+Our next example lists a directory tree recursively, and concurrently.
 
-import qualified Streamly.Prelude as Stream
-import qualified Streamly.Internal.FileSystem.Dir as Dir (toEither)
+This example uses the tree traversing combinator `parConcatIterate`.  This
+combinator maps a stream generator function on the input stream and then
+recursively on the generated stream as well and flattens the results. We map a
+directory to a stream generating its children and a file to a nil stream. This
+results in a concurrent recursive depth first traversal of the directory tree.
 
--- Lists a directory as a stream of (Either Dir File).
-listDir :: String -> SerialT IO (Either String String)
-listDir dir =
-      Dir.toEither dir               -- SerialT IO (Either String String)
-    & Stream.map (bimap mkAbs mkAbs) -- SerialT IO (Either String String)
+Please see [ListDir.hs][] for the complete working code.
```

Comment:
We should remember to update `ListDir.hs` accordingly. Make the examples consistent to what we described here.